storagenode/pieces/store.go

```diff
@@ -81,6 +81,12 @@ func NewStore(log *zap.Logger, blobs storage.Blobs) *Store {
 	}
 }
 
+// WithBlobs replaces the blob store with the provided blobs.
+func (store *Store) WithBlobs(blobs storage.Blobs) *Store {
```

Comment:
@egonelbre do you have a better idea how to do this?

Reply:
Reconfigure allows you to use a custom storage node db. There you can wrap the blobs as needed.

``` go
type SlowDB struct { storagenode.DB }
func (db *SlowDB) Pieces() storage.Blobs { return &SlowBlobs{db.Latency, db.DB.Pieces()} }
```

If you need it to depend on the call... you can pass that information via context... e.g.

```
// do regular stuff
ctx := context.WithValue(ctx, slowblobs.Enabled, true)
// now it's going to be slow
```

Reply:
Thanks! Indeed this would look better.

I am trying to reconfigure like this:

```go
testplanet.Run(t, testplanet.Config{
		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
		Reconfigure: testplanet.Reconfigure{
			NewStorageNodeDB: func(index int) (storagenode.DB, error) {
				return &SlowDB{
					DB: <db-to-wrap>,
					Latency: 110 * time.Millisecond,
				}, nil
			},
		},
	}
```

Where am I supposed to get `<db-to-wrap>` from? This should be the storagenode.DB created by the testplanet, but I don't find a way to acces it.

Reply:
`NewStorageNodeDB` is the func that creates the database. So you would need to call `storagenodedb.New` manually. It might actually make sense to give the "default db constructor" as an argument to `NewStorageNodeDB`.

Reply:
@egonelbre I am trying to follow the idea with the `context.WithValue`, but it seems it won't work in this test case.

The issue is that I want the blob store to be fast until I want to download the shares. At this moment I derive a child context with value `slowblobs.Enabled` and pass with to `verifier.DownloadShares`.

Unfortunately, `verifier.DownloadShares` is exectued on the satellite. The respective storage nodes is not affected at all by this derived context.

Reply:
Ah, right... yeah... then in that case I would keep a references to the created slow blob stores and call "enable", or something, before doing the slow test.

Reply:
Cool. This worked: https://github.com/storj/storj/pull/2210/commits/d976aff07fd682d15de8e4bd2c8158843e19a103